### ISSUE TITLE
Cart UI Polish Part 2

### ISSUE THREAD

**Issue by valeriegohier** [Shopify/dawn#206]
A few points of feedback / issues to address.  The cart page should be revisited holistically, while considering the following points:

- [x] Images look small. It's almost hard to see the products
- [x] The stroke around the ''garbage'' draws lots of attention to this action, especially for a cart.
![Screen Shot 2021-07-15 at 9 08 18 PM](https://user-images.githubusercontent.com/53573494/125877232-1a1853ec-e6fc-49b0-ac2d-087a53eaa50b.png)
- [x] Alignment of the different components seems off 
![image](https://user-images.githubusercontent.com/8127145/126669777-405b3b60-94ba-401f-a5a3-30e5896166cc.png)
- [x] There is a lot of white space between line items and the subtotal. 
https://screenshot.click/21-57-3ljwg-lg4ru.png

**Comment by PaulNewton** [comment]
>-Images look very small. It's almost hard to see the products
This will vary widely among merchant photography,  in the example screen the product content photography is shot as if just beyond arms reach and **in portrait**, with a similar background increase visual depth. 
 If this was made by default to seem bigger by either cropping, changing aspect ratios, or <img> width|height then merchants with closer product photography could get the opposite of photos looking too big.

Any change should tested against 3 scales: micro macro mega (coins, people, boats for example)

A settingless solution would be great but might be difficult given the restraints  #1 cart page information being in <table> and #2 product photography inconsistences show most apparently in the carts. The <table> restraint can make it hard to do things like letting the image be part of each items background to get more display area.

With a setting solution might want to treat similar to the common slideshow shopify solution to handle cropping , https://www.shopify.com/partners/blog/shopify-slideshow.


> -The stroke around the ''garbage'' draws too much attention to this action (especially for a cart!)
@valeriegohier  Should this be a configurable setting or tied to another color|typography option like an accent or themewide border settings. If so what are some candidate settings?

**Comment by melissaperreault** [comment]
## UX Update

I will get back to you @tauthomas01 about the quantity UI change, but you can follow [this layout for the update](https://screenshot.click/21-08-a6qk6-knh4c.png):

- [ ] Align left the Quantity column with the `Subtotal` and `Checkout` button to the left
- [ ] Remove completely the Price column but make sure to have the Total column display the sale price and the compare-at-price shown.
- [ ] Reduce the compare-at-price font size (let's chat to see our options)
- [ ] Reduce variant line items font size (let's chat to see our options)
- [ ] Increase the product thumbnail to 100px width at least, I'll check when reviewing. This would be for large screens, let's adjust together for smaller screens

**Comment by tauthomas01** [comment]
Hello @melissaperreault 

> Align left the Quantity column with the Subtotal and Checkout button to the left

This might be tricky to fix because the cart footer (subtotal section) is not part of the main cart so matching the layout for both cart body and cart footer can get complicated.

> Remove completely the Price column but make sure to have the Total column display the sale price and the compare-at-price shown.

Currently, when we have a “compare to price” price, we only show the final price, not the original price. Do we also want to show the original price? In contrast, when there is a discount, we see the price before the discount + after the discount.

Also, if we move the discount price to the Total column, the price is reflecting one unit (1 quantity). Is that what we want?

[![alt](https://screenshot.click/09-40-c2spp-esoam.png)](https://screenshot.click/09-40-c2spp-esoam.png)

> Reduce the compare-at-price font size (let’s chat to see our options)
> Reduce variant line items font size (let’s chat to see our options)

Let me know when you want to chat!

**Comment by melissaperreault** [comment]
> This might be tricky to fix because the cart footer (subtotal section) is not part of the main cart so matching the layout for both cart body and cart footer can get complicated.

Is this due to the fact that they are 2 separate sections?


> Currently, when we have a “compare to price” price, we only show the final price, not the original price. Do we also want to show the original price? In contrast, when there is a discount, we see the price before the discount + after the discount.

Oh this is true, I didn't express myself well then! Good catch!

We will remain how we do not display the compare at price in the cart like we didn't before and just display the discount price like shown. We also what the total column to reflect the multiplier if the quantity is more than 1 for instance. Does it make more sense? Lets chat during our pairing time tomorrow just making sure I understand the details!

**Comment by melissaperreault** [comment]
## UX update

### Update on the quantity UI component

Visual reference for the outcome for all screens.

- [ ] We'll use a select component with a border opacity decreased in the cart template (8% of text). The remove can icon will be displayed beside the dropdown, on the right.
- [ ] The dropdown will use the native list for all devices
- [ ] We'll need to add a setting to set the maximum number merchant want their buyers to purchase. This setting would apply to all line item.

**Comment by melissaperreault** [comment]
> Is there a rationale doc for this change, or any usability testing done?

@svinkle I hear your concerns and they are very legitimate. We based our decisions on various criteria:

- Reduce visual noise to align with the minimalist principle for Dawn. The + and the - of the quantity input was unpleasing and crowded when combined with the remove trash icon. We received strong feedback against the boxiness of the input and after exploring many variations, we didn't conclude it was the right UI compare to the select one.
- We were able to identify most of the individual units in an order was below 20.
- [Baymard secondary research](https://baymard.com/blog/drop-down-usability): _"One type of drop-down input where it’s acceptable (even preferable) to use a drop-down, even though there are many more than 10 options, are numerical inputs (e.g., [expiration date fields](https://baymard.com/blog/how-to-format-expiration-date-fields) and **product quantity fields**). In testing we observe users interacting with these drop-down inputs don’t have the same usability issues as users interacting with text-based drop-down inputs."_ - From our benchmark, most of the brands were offering the dropdown UI with a limit up to 10 or 20. This seemed aligned with what we observed in our numbers. 



What we decided to go with that you will find in the next issue to change the UI across Dawn is:

- [ ] Add a new number setting type to let merchants adjust the value themselves to match their business needs. (Learn how merchants are using it.) (Depending on the limitation we may need to go back to a range setting, still investigating. But regardless we are aiming to put a max.)
- [ ] Have this number setting maxed at 500 to accommodate larger merchants needs.

I hope this helps understand why we took this direction. We're available for additional chats if you have additional concerns. 🙏

**Comment by PaulNewton** [comment]
tl;dr this is putting a bound on something meant to fly 📈 .
💣💥 ⚰️, Careful with settings defaults of max quantities,  a universally applied `max` attribute ,or opinionated upper bound, on quantity inputs is SILENTLY poison to merchants businesses, leak on a spaceship and all that.
I'd avoid doing this here **upstream** or at least in tandem with the `<select>` push without clear consideration of warning labels on such things or an optin.
It's right up there with settings to turn off the AtC or checkout: accidental toggling costs money. Though sections everywhere is shipping that now as standard it seems 🤷 ..
Merchants wont always know in a timely fashion **why** their AOV is low when they need volume if there is a max _default_ buried in the theme code or a setting, or a long forgotten template using a dynamic source from an off page metafield definition.

It is the themes job to facilitate boundless growth there should be friction putting in this specific type of restriction,  not easy or at least not easy to ever forget it has been done for the general use case even if some industries have this as a requirement.
On the other side of things merchants need to be better informed of how easy it is to bypass such artificial limits on the OSC and that currently only apps can effectively manage purchase limits automatically.

>We'll use a select component

Using **only** select for quantity input is setting up downstream themes for a lot of pain, _unless you mean as an underlying hidden element as the source of truth like a variant selector_.

>Have this number setting maxed at 500 to accommodate larger merchants needs.

Have you ever scrolled through a select with 500 entries? it's annoying enough doing 50 for state selectors in the US and ~100 for year/age checks.
Now do it more than 10 times for a B2B purchase.

🤓 BUT 500 is easy peasy  squeezy for keyboard warriors using type ahead <kbd>5</kbd> -> <kbd>0</kbd> -> <kbd>0</kbd>, but not so much for everyone else: the mousers, the scrollers, the touchers, the arrowers , and the speakers.
>[Drop-downs quickly become difficult for users when they are presented with an overwhelming number of options to choose from](https://baymard.com/blog/drop-down-usability#:~:text=Drop-downs%20quickly%20become%20difficult%20for%20users%20when%20they%20are%20presented%20with%20an%20overwhelming%20number%20of%20options%20to%20choose%20from) -baymard

**Logistically**  ,aside from the accessibility minutia ,  a `<select>` for line_item quantities will become a tangle that will have to get _untangled_ by someone and represents a much bigger cliche anti-pattern than <kbd>-[]+</kbd> inputs: _trying to make a singular universal human interface input._ 

**Researchically**,  isn't the reference research misinterpreted? it is **specifically** comparing numbers in `<select>` vs **TEXT** in `<select>` and NOT numeric text inputs `<input type="number">` compared against `<select>` used for _numerical_ choice.
DRASTIC difference.
>[users interacting with these _drop-down_ inputs don’t have the same usability issues as users interacting with _text-based drop-down inputs_](https://baymard.com/blog/drop-down-usability#:~:text=users%20interacting%20with%20these%20drop-down%20inputs%20don%E2%80%99t%20have%20the%20same%20usability%20issues%20as%20users%20interacting%20with%20text-based%20drop-down%20inputs) -baymard.com

**Aesthetically** I get  it ,the allure of fixing the current doctrine of fake spinner tri-input boxes <kbd>-[]+</kbd> , that I personally consider an antipattern UI and would be a boon for a lot of theme devs to never ever **ever** have to touch,tweak,implement or fix the wicked things again.

> most of the individual units in an order was below 20

Keyword "most" , for every other merchant that doesn't fit that AOQ slice of the stats spectrum imagine their shock why the item they need to sell in high volume has been performing poorly: because shopify base themes arrogantly decided **years ago** to put a `max`  on how many things a customers could buy for aesth-tech.
I.E only has 5% market share but if a businesses demographic uses that browser suddenly words like "most" become very costly.

And any universally applied `max` attribute on quantity inputs is SILENTLY risky. < --- prolly biggest point of risk in this entire issue page probably, and i've buried the lead sorry moved this point to the opening paragraph, .


Expiry years for credit cards are reasonably bounded, not so for a merchants possible inventory quantity they want immediately accessible to customer choice without having to make a draft-order because shopify themes have become fundamentally flawed for convenience of aesthetic implementation.
Using only a `<select>` in the base reference and nothing else is not the approach for an ecommerce UI paradigm intended to be **_unbounded_** either by reality of logistics, hopeful capitalism, or end user expectation to be able to GIVE the amount of quantity they want of a product across web and retail interactions.

An exception in all this of course is for managed inventory of reasonable stock,  _under the proposed thresholds_, because then the intent cannot be unbounded as their is a **KNOWN** real world inventory the merchant has **chosen** to control in that way.

To leave triput <kbd>-[]+</kbd> quantities in the past what we may be looking at a need for an adapative UI pattern or merchant block choice.
- 1-5 range slider if setting, managed inventory , or block choice ==  `<input type="range" id="qty" name="quantity" min="0" max="5">`
- 1-20 if managed inventory == `<select>`
- 20+ or unmanaged inventory  == `<input type="number">`
but that comes with the problem of consistency across line_item factors of availability and inventory mgmt.

Alternatives are hybrids where unmanaged inventory gets a _manual entry_ link under the `<select>` providing a text input popup|reveal ala polaris admin inventory popups.(todo add screenshot).

🤔 Or that when a customer chooses a line_items `<select>` max value that sets a line item property which allows them to add another line_item of that product, or a 0 quantity line_item duplicate is added for them automatically to choose the new overlow amount;  but that undermines any  concept of "max purchase amount". 🤕 

Having a manual input fallback either as a default or merchant choice seems a hard reality.



Really like seeing the research references in the build process of the reference theme though 🎉

**Comment by melissaperreault** [comment]
## UX Update

After considering the pros and cons of both dropdown and input UI, we'll revert to the [input UI](https://screenshot.click/21-08-00g76-7w715.png) to enable all merchants need.

- [ ] Display the quantity input UI but the border opacity at 8%
- [ ] We can keep the spacings between the box and the trash icon
- [ ] We no longer need the theme setting to set a max, we can remove it
